python/pyspark/mlv2/io_utils.py

```diff
@@ -0,0 +1,159 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import os
+import time
+from typing import Any, Dict, Optional
+from pyspark.errors import PySparkNotImplementedError, PySparkRuntimeError
+from pyspark.util import VersionUtils
+
+from pyspark import __version__ as pyspark_version
```

Comment:
on python client, this version info may be different with the one `spark.version` from the sever side

Reply:
For new pyspark ML module, I think we should regard it as a client side package so using client side pyspark version should be fine. Pyspark server side just runs wrapped python UDF which is unaware of concrete ML algorithm logic.